pkgs/by-name/cl/clash-verge-rev/package.nix

```diff
@@ -1,66 +1,131 @@
-  ];
-
-  runtimeDependencies = [
-    (lib.getLib udev)
-    libayatana-appindicator
-  ];
+  src-service = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-service";
+    rev = "e74e419f004275cbf35a427337d3f8c771408f07"; # no meaningful tags in this repo. The only way is updating manully every time.
+    hash = "sha256-HyRTOqPj4SnV9gktqRegxOYz9c8mQHOX+IrdZlHhYpo=";
+  };
 
-  installPhase = ''
-    runHook preInstall
+  service-cargo-hash = "sha256-tjPGXsABNK3HR6QmVl97UqGkDBhqhL3sSc3vGScWmho=";
 
-    mkdir -p $out/bin
-    mv usr/* $out
+  service = callPackage ./service.nix {
+    inherit
+      version
+      rustPlatform
+      src-service
+      pkg-config
+      openssl
+      webkitgtk
+      service-cargo-hash
+      pname
+      ;
+  };
 
-    runHook postInstall
-  '';
+  webui = callPackage ./webui.nix {
+    inherit
+      version
+      src
+      pname
+      pnpm
+      nodejs
+      stdenv
+      ;
+  };
 
-  postFixup = ''
-    rm -f $out/bin/verge-mihomo
-    ln -sf ${lib.getExe clash-meta} $out/bin/verge-mihomo
-  '';
+  sysproxy-hash = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
 
-  passthru.updateScript = nix-update-script { };
+  unwrapped = callPackage ./unwrapped.nix {
+    inherit
+      pname
+      version
+      src
+      sysproxy-hash
+      webui
+      meta
+      makeDesktopItem
+      pkg-config
+      rustPlatform
+      openssl
+      webkitgtk
+      lib
+      nix-update-script
+      ;
+  };
 
   meta = {
     description = "Clash GUI based on tauri";
     homepage = "https://github.com/clash-verge-rev/clash-verge-rev";
     license = lib.licenses.gpl3Plus;
     mainProgram = "clash-verge";
-    maintainers = with lib.maintainers; [ Guanran928 ];
+    maintainers = with lib.maintainers; [
+      Guanran928
+      bot-wxt1221
+    ];
     platforms = [ "x86_64-linux" ];
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
   };
-})
+in
+stdenv.mkDerivation {
+  inherit
+    pname
+    src
+    version
+    meta
+    ;
+  nativeBuildInputs = [
+    makeShellWrapper
+    wrapGAppsHook3
+  ];
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,share,lib/clash-verge/resources}
+
+    cp -r ${unwrapped}/share/* $out/share
+    ln -s ${service}/bin/clash-verge-service $out/bin/clash-verge-service
+    cp -r ${unwrapped}/bin/clash-verge $out/bin/clash-verge
+    ln -s ${mihomo}/bin/mihomo $out/bin/verge-mihomo
+    # people who want to use alpha build show override mihomo themselves. The alpha core entry was removed in clash-verge.
+    ln -s ${v2ray-geoip}/share/v2ray/geoip.dat $out/lib/clash-verge/resources/geoip.dat
+    ln -s ${v2ray-domain-list-community}/share/v2ray/geosite.dat $out/lib/clash-verge/resources/geosite.dat
+    ln -s ${dbip-country-lite.mmdb} $out/lib/clash-verge/resources/Country.mmdb
+
+    wrapProgram $out/bin/clash-verge \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libayatana-appindicator ]}"
```

Comment:
Since we build it from source we can patch the source directly instead of using a wrapper. E.g.

```
    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
      --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
```